multicluster/apis/multicluster/v1alpha1/resourceexport_webhook.go

```diff
@@ -53,8 +53,13 @@ func (r *ResourceExport) Default() {
 	if kindLabelVal, exists := r.Labels[constants.SourceKind]; !exists || kindLabelVal != constants.AntreaClusterNetworkPolicyKind {
 		r.Labels[constants.SourceKind] = constants.AntreaClusterNetworkPolicyKind
 	}
-	if r.DeletionTimestamp.IsZero() && !stringExistsInSlice(r.Finalizers, constants.ResourceExportFinalizer) {
-		r.Finalizers = append(r.Finalizers, constants.ResourceExportFinalizer)
+	// Add domain qualified finalizer for ResourceExports to avoid Kubernetes from reporting errors:
+	//  "prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers"
+	// https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers
+	// Note for ResourceExports created before Antrea v2.2, LegacyResourceExportFinalizer may still be present
+	// and needs to be removed before a ResourceExport is deleted.
+	if r.DeletionTimestamp.IsZero() && !stringExistsInSlice(r.Finalizers, constants.DomainQualifiedResourceExportFinalizer) {
```

Comment:
not introduced by your PR but `stringExistsInSlice` is duplicated from `multicluster/controllers/multicluster/common/helper.go`, and nowadays it can easily be replaced by https://pkg.go.dev/slices#Contains (no need for a helper function anymore).